ocs_ci/ocs/resources/pod.py

```diff
@@ -1468,13 +1471,26 @@ def verify_pods_upgraded(old_images, selector, count=1, timeout=720):
                     f"Number of found pods {pods_len} is not as expected: " f"{count}"
                 )
             for pod in pods:
-                verify_images_upgraded(old_images, pod.get())
+                pod_obj = pod.get()
+                verify_images_upgraded(old_images, pod_obj)
```

Comment:
this check alone I feel is sufficient to check if the images got upgraded and are actually same, checking if they are exactly same will be repetitive.

Reply:
It's not, that's why I am doing following check to make sure all containers have the same image.
Because we added this logic:
https://github.com/red-hat-storage/ocs-ci/pull/4733
Because of one bug that some pod doesn't need to restart if image has the same sha can lead that we are just ignoring the images which has the same sha but different image source quay.io/registry.redhat.io .  So we think of that OSD images are ok and that everything is upgraded but OCS upgrade process is not yet finished because in case of OSDs it's restarting pods anyway.  If we do not wait till all pod containers have the same image then we leave this section and verifying the OCS status as another step in upgrade test which leads to weird issues because pods are still restarting.

This logic will prevent us to go further in verification till we are sure all pods are really restarted and upgraded.